common/modules/nft_deposit/src/nft_deposit.rs

```diff
@@ -0,0 +1,145 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::{GenericEsdtAmountPair, Nonce};
+
+#[elrond_wasm_derive::module]
+pub trait NftDepositModule: token_send::TokenSendModule + token_supply::TokenSupplyModule {
+    fn deposit_tokens(
+        &self,
+        payment_token_id: TokenIdentifier,
+        payment_token_nonce: Nonce,
+        payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        require!(payment_amount != 0, "Cannot deposit 0 tokens");
+        require!(payment_token_nonce != 0, "Cannot deposit fungible tokens");
+        require!(
+            self.nft_deposit_accepted_token_ids()
+                .contains(&payment_token_id),
+            "Not an accepted token id"
+        );
+
+        let payment = GenericEsdtAmountPair {
+            token_id: payment_token_id,
+            token_nonce: payment_token_nonce,
+            amount: payment_amount,
+        };
+
+        let mut index = 1;
+        let mut entry_updated = false;
+        let caller = self.blockchain().get_caller();
+        let deposit_len = self.nft_deposit(&caller).len();
+
+        while index <= deposit_len {
+            let mut entry = self.nft_deposit(&caller).get(index);
+
+            if self.equal_token_type(&entry, &payment) {
+                entry.amount = &entry.amount + &payment.amount;
+                self.nft_deposit(&caller).set(index, &entry);
+                entry_updated = true;
+                break;
+            }
+
+            index += 1;
+        }
+
+        if !entry_updated {
+            require!(
+                deposit_len + 1 < self.nft_deposit_max_len().get(),
```

Comment:
Replaced with <=. Was < because the deposit of max 10 would be merged with one more token that was not yet created and will result in a 11 tokens merge, but that shouldn't be a problem

Reply:
ok